src/Packs.sol

```diff
@@ -171,7 +170,7 @@ contract Packs is
         PackType packType_,
         BucketData[] memory buckets_,
         bytes memory signature_
-    ) public payable whenNotPaused returns (uint256) {
+    ) external payable whenNotPaused returns (uint256) {
```

Comment:
Yeah, ok, for the `calldatacopy` external will make a non-trivial difference because the bucket array won't need to be written to memory. Otherwise we're talking like < 100 gas.